### PR TITLE
openssl 1.0.2f

### PR DESCRIPTION
**Pull request by vszakats**

### CODE REVIEW

File: Library/Formula/openssl.rb [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
   homepage "https://openssl.org/"
-  url "https://www.openssl.org/source/openssl-1.0.2e.tar.gz"
-  mirror "https://dl.bintray.com/homebrew/mirror/openssl-1.0.2e.tar.gz"
-  mirror "https://www.mirrorservice.org/sites/ftp.openssl.org/source/openssl-1.0.2e.tar.gz"
-  sha256 "e23ccafdb75cfcde782da0151731aa2185195ac745eea3846133f2e05c0e0bff"
+  url "https://www.openssl.org/source/openssl-1.0.2f.tar.gz"
+  mirror "https://dl.bintray.com/homebrew/mirror/openssl-1.0.2f.tar.gz"
+  mirror "https://www.mirrorservice.org/sites/ftp.openssl.org/source/openssl-1.0.2f.tar.gz"
+  sha256 "932b4ee4def2b434f85435d9e3e19ca8ba99ce9a065a61524b429a9d5e9b2e9c"
   revision 1
```
Comment (DomT4) [inline Library/Formula/openssl.rb]: This line needs removing.

**Comment by DomT4** [comment]
They haven't even officially announced this as released on the mailing list yet; nice spot! :smile:

**Comment by vszakats** [comment]
:) yep!

(will correct the `revision` line!)

**Comment by vszakats** [comment]
[ Let's wait for the formal announcement to avoid missing a last minute update/re-upload. ]

**Comment by DomT4** [comment]
Don't worry, this isn't going to get done before they release any kind of formal announcement. Even if we didn't want to wait, the current CI queue and how long OpenSSL builds take to check everything else mandates it :wink:.

**Comment by alex** [comment]
Formal announcement is out: https://openssl.org/news/secadv/20160128.txt

**Comment by vszakats** [comment]
Found the first regression: https://github.com/openssl/openssl/issues/595

**Comment by alex** [comment]
In addition to the regression, tests are failing because the OpenSSL website is having some availability issues.

**Comment by DomT4** [comment]
The OpenSSL website is currently dying heavily under what I'd presume is traffic load, so this might even fail to fetch when CI processes it given mirrors haven't updated yet.

**Comment by DomT4** [comment]
Heh, @alex beat me to it.

**Comment by jomo** [comment]
plain http to the rescue! :sob:

**Comment by vszakats** [comment]
Uploaded copies here:
https://transfer.sh/1ZTDt/openssl-1.0.1k.tar.gz
https://transfer.sh/81UZp/openssl-1.0.2f.tar.gz

**Comment by jomo** [comment]
http and ftp are also working from here:
- ftp://ftp.openssl.org/source/openssl-1.0.2f.tar.gz
- http://www.openssl.org/source/openssl-1.0.2e.tar.gz

**Comment by DomT4** [comment]
You couldn't pay me enough money to download OpenSSL over an insecure connection :wink:. I had a copy in my cache from earlier; uploading to the brew mirror.

**Comment by jomo** [comment]
I wasn't aware sha256 has been broken :D

**Comment by vszakats** [comment]
Agreed @DomT4!

**Comment by vszakats** [comment]
@jomo I generated that SHA256 after downloading the package via HTTPS.

**Comment by jomo** [comment]
I grabbed the sha256 from the openssl.org server and compared it to my http download.

**Comment by DomT4** [comment]
> I wasn't aware sha256 has been broken :D

It hasn't, but why take unnecessary security & privacy risks? All unencrypted traffic is inherently risky, however small or large that risk is :smile:.

File has been uploaded to the brew mirrors.

**Comment by vszakats** [comment]
Luckily it matches the one announced, which was hopefully generated on the build machine (and included in the announcement as a signed text) :) From that point HTTP is fine!

**Comment by DomT4** [comment]
@vszakats We may want to backport https://github.com/openssl/openssl/commit/41a28cb2944a4e1c9d13889757a3bd9f72abeca1 here.

**Comment by alex** [comment]
It looks like there's a 2nd issue which may necessitate a re-release; https://mta.openssl.org/pipermail/openssl-dev/2016-January/004484.html

**Comment by vszakats** [comment]
@DomT4 Added my patch.
@alex I'd appreciate a ping here when this issue is addressed.

**Comment by DomT4** [comment]
Let's give this till tomorrow afternoon to see if OpenSSL move upstream on the remaining issue. If they don't we may have to merge this and patch whatever they do with that problem afterwards.

**Comment by vszakats** [comment]
Agreed, let's wait.

**Comment by DomT4** [comment]
@vszakats Fixed upstream. We probably want to pull in https://github.com/openssl/openssl/commit/7107798ae6c5e19f581915928a69073d17cc21ab & https://github.com/openssl/openssl/commit/83ab6e55a1f8de9b3e45d13dcc78eb739dc66dea and _then_ we should be good to roll this.

**Comment by vszakats** [comment]
Thanks. Included these new patches and squashed.

**Comment by DomT4** [comment]
Thanks for the quick response! I've kicked you off the build queue for a couple minutes to update Jenkins but I'll stick you back on very shortly after.

**Comment by DomT4** [comment]
@BrewTestBot test this please

**Comment by DomT4** [comment]
Hmm. It can't find the first patched file, the one that you had inline yesterday and worked. Bit odd:

```
==> Verifying openssl--patch-98443034f57e5c4fd1bd89dbf64e9b150184522d10b6a6f7bb7e67cc397615c2.diff checksum
==> Patching
/usr/bin/patch -g 0 -f -p1 -i /private/tmp/openssl--patch20160129-31839-ovlbi3/41a28cb2944a4e1c9d13889757a3bd9f72abeca1.diff
can't find file to patch at input line 5
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/include/openssl/bio.h b/include/openssl/bio.h
|index ea1d3b7..beacf19 100644
|--- a/include/openssl/bio.h
|+++ b/include/openssl/bio.h
--------------------------
No file to patch.  Skipping patch.
1 out of 1 hunk ignored
Error: Failure while executing: /usr/bin/patch -g 0 -f -p1 -i /private/tmp/openssl--patch20160129-31839-ovlbi3/41a28cb2944a4e1c9d13889757a3bd9f72abeca1.diff
```

**Comment by vszakats** [comment]
Indeed, I took that patch from the master branch, that means 1.1.0. Let me revise that.

**Comment by vszakats** [comment]
Fixed!

**Comment by DomT4** [comment]
The CI failures are wholly unrelated to the OpenSSL change here, so merged in https://github.com/Homebrew/homebrew/commit/2e351f745b65184cd933613ef27ed72c0ff3c6c2.

Thank you both @alex and @vszakats for chasing OpenSSL regressions and the speedy responses, and to @vszakats for submitting the PR! Appreciate the cooperation throughout! :beers:

**Comment by vszakats** [comment]
Thank you @DomT4. (I was about to bump revision for `libevent`, but I'm glad it's fine without it.)

**Comment by DomT4** [comment]
Yeah the `spdylay` test failure is happening because `libevent` gets unlinked earlier in the build to resolve one `conflicts_with` conditional and then isn't relinked after that conflict has passed. We're aware of the issue.

**Pull request closed without merging**